src/event/format/json.rs

```diff
@@ -225,3 +231,10 @@ fn valid_type(data_type: &DataType, value: &Value, schema_version: SchemaVersion
         }
     }
 }
+
+pub fn is_parsable_as_number(value: &Value) -> bool {
+    let Value::String(s) = value else {
+        return false;
+    };
+    s.parse::<f64>().is_ok() || s.parse::<i64>().is_ok()
```

Comment:
```suggestion
    s.parse::<f64>().is_ok()
```
all strings containing numbers are parsable as f64